source/server/admin/clusters_params.cc

```diff
@@ -0,0 +1,29 @@
+#include "source/server/admin/clusters_params.h"
+
+#include "envoy/http/codes.h"
+#include "envoy/http/query_params.h"
+
+#include "absl/strings/string_view.h"
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Server {
+
+Http::Code ClustersParams::parse(absl::string_view url, Buffer::Instance& response) {
+  UNREFERENCED_PARAMETER(response);
+
+  Http::Utility::QueryParamsMulti query =
+      Http::Utility::QueryParamsMulti::parseAndDecodeQueryString(url);
+  absl::optional<std::string> optional_format = query.getFirstValue("format");
+  if (optional_format.has_value()) {
+    if (*optional_format == "json") {
+      format_ = Format::Json;
+    } else {
+      format_ = Format::Text;
```

Comment:
ah -- should we fail if optional_format is not empty or "text"?

if not, could you comment why?

Reply:
From a developer experience perspective, it would seem to me that defaulting to some reasonable format option that will succeed _could_ be helpful. So, if the user makes a mistake, they at least get some data that is useful. This is only really relevant in the case of an ad-hoc query of course.

If this isn't consistent with the user experience in other handlers I am open to returning the appropriate failure response.

Reply:
Upon further thought, it might make more sense to return an error so that systems depending on it, such as Istio, can respond appropriately in that case. I'll update it.